android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt

```diff
@@ -51,7 +51,8 @@ class MusicService : HeadlessJsTaskService() {
     private var notificationCapabilities: List<Capability> = emptyList()
     private var compactCapabilities: List<Capability> = emptyList()
 
-    fun setupPlayer(playerOptions: Bundle?, promise: Promise?) {
+    @MainThread
```

Comment:
All Android services run in the main thread, so this is not needed I think. Was there a specific reason for these annotations?

Reply:
I added these at the request of @dcvz , I believe the thought was to notify people looking at the code that those methods would be on the main thread.

Happy to remove if you give me the clearance.

Reply:
so we're no longer using `scope.launch` inside the Service but rather at the Module level -- so annotating them with `@MainThread` is just so the IDE knows that when the Module calls them.. it should be within a main thread scope.

Reply:
The reason for this is that at the module level if you did something like:
```kotlin
// pseudo code
player.add(tracks) // this gets called at the service level in a coroutine.. so the next line will be called immediately without actually waiting for the add method to be called.. 
promise.complete()
```